woocommerce/wc-admin
# WooCommerce Shipping order page banner prompt

Implements a banner that is displayed in the WooCommerce order page, whenever *all* these conditions are met:

- WooCommerce Shipping banner NOT dismissed
- Jetpack >= 4.4 installed and active
- Jetpack Connected
- No incompatible extensions installed:
   - Shipstation not installed.
   - UPS not Installed.
   - Fedex not installed.
   - ShippingEasy not installed.
- Order contains physical products which need to be shipped ([we should check that the order status is not set to complete](https://a8c.slack.com/archives/CKNC7KACA/p1584395171186200)).
- Store is located in US.
- Store currency is set to USD.
- `WCS plugin not installed` OR` WCS is installed *AND* ToS have NOT been accepted *AND* WCS version is 1.22.5 or greater`.
    - The 1.22.5 or greater requirement is so we can launch the shipping modal from the banner.

### Screenshots

https://www.dropbox.com/s/m8dpajr0w1lyo8v/shot-2020-03-20-15-31-32.png?dl=0
https://www.dropbox.com/s/7mkrlxk5tnva29a/shot-2020-03-20-15-31-02.png?dl=0

### Detailed test instructions:

1. Setup your environment considering all the conditions described in this PR's summary, or to make it easy:
   1. Start with a fresh WooCommerce site (WooCommerce Services not installed)
   2. Install and Connect Jetpack
   3. Install `shipping-bannner` branch of wc-admin
2. Create an order with a shippable product.
3. Go to that order's page (**Note: the A/B testing logic only shows the prompt to 25% of sites. Set to the option `woocommerce_shipping_prompt_ab` to `b` to enable it for your test site**)
4. Press the create shipping label button.
5. The WooCommerce Services create shipping label modal should pop up, and a new label should be able to be purchased.


### Changelog Note:
